packages/trader/src/Modules/Trading/Components/Form/WheelPicker/wheel-picker.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react';
+import { Button, Icon, Text } from '@deriv/components';
+import { motion } from 'framer-motion';
+
+const variants = {
+    enter: (direction: 'up' | 'down') => ({
+        x: 0,
+        y: direction === 'down' ? 30 : -30,
+        opacity: 0,
+        transition: {
+            duration: 0.24,
+            ease: [0, 0, 0, 1],
+        },
+    }),
+    center: {
+        x: 0,
+        y: 0,
+        opacity: 1,
+        transition: {
+            duration: 0.24,
+            ease: [0, 0, 0, 1],
+        },
+    },
+    exit: (direction: 'up' | 'down') => ({
+        x: 0,
+        y: direction === 'down' ? -30 : 30,
+        opacity: 0,
+        transition: {
+            duration: 0.24,
+            ease: [0, 0, 0, 1],
+        },
+    }),
+};
+
+type WheelPickerType = {
+    onClick: (id: number) => Promise<void> | void;
+    options: number[];
+    defaultValue?: number;
+    currency?: string;
+};
+
+const WheelPicker = ({ options, onClick, defaultValue, currency }: WheelPickerType) => {
+    const [selectedIndex, setSelectedIndex] = useState(Math.floor(options.length / 2));
+    const [direction, setDirection] = useState('down');
+
+    useEffect(() => {
+        if (defaultValue) {
+            setSelectedIndex(options.indexOf(defaultValue));
+        }
+    }, [defaultValue, options]);
+
+    const handleIndexChange = (newIndex: number, newDirection: 'up' | 'down') => {
+        setDirection(newDirection);
+        setSelectedIndex(newIndex);
+        onClick(options[newIndex]);
+    };
+
+    const handleIncrease = () => {
+        if (selectedIndex > 0) {
+            handleIndexChange(selectedIndex - 1, 'up');
+        }
+    };
+
+    const handleDecrease = () => {
+        if (selectedIndex < options.length - 1) {
+            handleIndexChange(selectedIndex + 1, 'down');
+        }
+    };
+
+    const visibleValues = () => {
+        const values = [];
+        if (selectedIndex > 0) {
+            values.push(options[selectedIndex - 1]);
+        } else {
+            values.push('');
+        }
+        values.push(options[selectedIndex]);
+        if (selectedIndex < options.length - 1) {
+            values.push(options[selectedIndex + 1]);
+        } else {
+            values.push('');
```

Comment:
Sorry for silly question, do we need this empty strings?

Reply:
Yes , When we move the picker to the top where there are only 2 values of picker we have to show the empty space on bottom or top (in case the picker is moved to bottom) . The space other then 2 values is managed by this string